tests/View/Blade/BladePhpStatementsTest.php

```diff
@@ -141,4 +141,35 @@ public function testUnclosedParenthesisForBladeTags()
 
         $this->assertEquals($expected, $this->compiler->compileString($string));
     }
+
+    public function testNestedTagCalls()
+    {
+        $string = "<span @class(['k' => @empty(\$v)])></span>";
+        $expected = '<span class="<?php echo \Illuminate\Support\Arr::toCssClasses([\'k\' => @empty($v)]) ?>"></span>';
```

Comment:
I don't understand - why isn't the `@empty` compiled. What was the behavior before your original Blade PR?

Reply:
It is because now the regex eagerly captures the first ')' it sees, and continues on to the following ")" if the first one was not the right one.
In this example:
```php
@class(['k' => @empty($v1), 't' => @empty($v2)])
```
regex captures the `@class(['k' => @empty($v1` and `@empty($v2` as the `$matches` variable. Then starts to inspect the first one. (`@class(...`) While inspecting the `@class it add forward the rest of the template to reach the correct closing ")" and will result in :
`@class(['k' => @empty($v1), 't' => @empty($v2)])`.
All good up until now.
But the second match of regex $matches[1] is now redundant (`@empty($v2)`) because it is covered by the while loop, so we unset it to avoid extra compilation.

Reply:
The original code also passes the same tests. I mean the @empty do not get compiled.
This is the original code before the fix:
```php  
  protected function compileStatements($value)
    {
        return preg_replace_callback(
            '/\B@(@?\w+(?:::\w+)?)([ \t]*)(\( ( (?>[^()]+) | (?3) )* \))?/x', function ($match) {
                return $this->compileStatement($match);
            }, $value
        );
    }
```